cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguageFrontend.kt

```diff
@@ -326,6 +324,112 @@ class PythonLanguageFrontend(language: Language<PythonLanguageFrontend>, ctx: Tr
         return tud
     }
 
+    private fun getRootPath(file: File, rootPaths: List<Path>): Path {
+        rootPaths.forEach {
+            if (file.path.removePrefix(it.toString()) != file.path.toString()) return it
+        }
+        return file.toPath().parent
+    }
+
+    override fun gatherExternalSources(
+        rootPaths: List<Path>,
+        source: File,
+        externalSources: MutableList<File>,
+        processedImports: MutableList<String>,
+    ): List<File> {
+
+        val pythonExternalSources =
+            externalSources
+                .filter { externalFile ->
+                    language.fileExtensions.any { externalFile.path.endsWith(it) }
+                }
+                .toMutableList()
+
+        val importRe = "(?m)^(?:from[ ]+(\\S+)[ ]+)?import[ ]+(\\S+)[ ]*\$".toRegex()
+        val allImportedSources: MutableList<File> = mutableListOf()
+        importRe.findAll(source.readText()).forEach { imp ->
+
+            // Only try to find file containing imports if we did not process this import so far
+            if (!processedImports.contains(imp.groupValues.get(0))) {
```

Comment:
I don't think this approach works. An `from foo ...` import can refer to different imports, depending on the location of the file you're parsing here.

Reply:
Imports are now analyzed based on the import handling in the `StatementHandler`. So from import is now also handled in the same way as we do for the generation of the import declaration.